src/control/server/instance_exec.go

```diff
@@ -31,32 +31,47 @@ import (
 
 	srvpb "github.com/daos-stack/daos/src/control/common/proto/srv"
 	"github.com/daos-stack/daos/src/control/server/ioserver"
+	"github.com/daos-stack/daos/src/control/system"
 )
 
 // IOServerRunner defines an interface for starting and stopping the
 // daos_io_server.
 type IOServerRunner interface {
-	Start(context.Context, chan<- ioserver.InstanceError) error
+	Start(context.Context, chan<- error) error
 	IsRunning() bool
 	Signal(os.Signal) error
 	Wait() error
 	GetConfig() *ioserver.Config
 }
 
-// IsStarted indicates whether IOServerInstance is in a running state.
-func (srv *IOServerInstance) IsStarted() bool {
+// isStarted indicates whether IOServerInstance is in a running state.
+func (srv *IOServerInstance) isStarted() bool {
 	return srv.runner.IsRunning()
 }
 
-// Start checks to make sure that the instance has a valid superblock before
-// performing any required NVMe preparation steps and launching a managed
-// daos_io_server instance.
-func (srv *IOServerInstance) Start(ctx context.Context, errChan chan<- ioserver.InstanceError) error {
+func (srv *IOServerInstance) format(ctx context.Context, recreateSBs bool) (err error) {
```

Comment:
Follow-on to my general comment... It seems like it would be a lot cleaner to move all of the format stuff here. The other stuff is left over from the old single-instance way of running things. The harness shouldn't really know all of the internal details and be reaching down into and through the instance.

Reply:
do you mean move the contents of doFormat into srv.doFormat?

Reply:
Yes, although I suggest naming it something more like `.FormatStorage()`. Even though they're in the same package and it's possible for the harness to use unexported methods, it would make things a bit clearer if we adhered to the convention that exported methods are intended to be used outside of the struct, and unexported methods are internal helpers. It seemed like you were already heading in that direction with some of the renaming, which I support.

Reply:
okay to do this as part of https://github.com/daos-stack/daos/pull/2484 ?